### PR TITLE
Allow use of empty string as a value for a Attribute

### PR DESCRIPTION
**Pull request by martin-pil**
As talked about in https://github.com/suds-community/suds/issues/107

No tests are affected by this change, all tests ran fine locally.

### CODE REVIEW

**Comment by phillbaker** [comment]
Thanks for the PR, please add a test that covers this behavior.

**Comment by martin-pil** [comment]
After looking into "tostr" simplified the code a bit, but I would say that the check must have been used to block empty strings as tostr will always return a string. Hope that this is still something that can be added to suds.

File: tests/test_mx_appender.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+class TestPrimitiveAppender:
+
+    @patch("suds.mx.appender.Appender.__init__", return_value=None)
+    def test_append_string(self, _):
+        parent = Mock()
+        content = MockContent("_TEST_TAG", "")
+        primitive_appender = PrimitiveAppender()
+        primitive_appender.append(parent, content)
+        parent.set.assert_called_once_with(content.tag.lstrip("_"), content.value)
```
Comment (phillbaker) [inline tests/test_mx_appender.py:50]: If feasible, would prefer to minimize the patching/asserting on call values. 

```
from suds.sax.element import Element
...
parent = Element('foo')
content = MockContent("_TEST_TAG", "bar")
primitive_appender = PrimitiveAppender(Mock())
primitive_appender.append(parent, content)
assert parent.get("TEST_TAG") == "bar"

parent = Element("foo")
content = MockContent("_TEST_TAG", "")
primitive_appender = PrimitiveAppender(Mock())
primitive_appender.append(parent, content)
assert parent.get("TEST_TAG") == ""
```

File: suds/mx/appender.py [hunk 2]
```diff
@@ -275,8 +274,7 @@ def append(self, parent, content):
         if content.tag.startswith('_') and isinstance(content.type, Attribute):
             attr = content.tag[1:]
             value = tostr(content.value)
-            if value:
```
[the remark this answers is not in the document]
Comment (martin-pil) [reply suds/mx/appender.py:278]: Previously it was sent on, only empty string was filtered out. But can check if string "value == 'None'" or if "content.value is None"

[the remark this answers is not in the document]
Comment (phillbaker) [reply suds/mx/appender.py:278]: Ah, thanks, makes sense.

File: tests/test_mx_appender.py (hunk 1 above)
Comment (phillbaker) [reply tests/test_mx_appender.py:50]: Can you take a look at updating this test?

**Pull request merged**